cmd/xl-v1-object.go

```diff
@@ -166,13 +166,21 @@ func (xl xlObjects) CopyObject(ctx context.Context, srcBucket, srcObject, dstBuc
 
 // GetObjectNInfo - returns object info and an object
 // Read(Closer). When err != nil, the returned reader is always nil.
-func (xl xlObjects) GetObjectNInfo(ctx context.Context, bucket, object string, rs *HTTPRangeSpec, h http.Header) (gr *GetObjectReader, err error) {
+func (xl xlObjects) GetObjectNInfo(ctx context.Context, bucket, object string, rs *HTTPRangeSpec, h http.Header, writeLock bool) (gr *GetObjectReader, err error) {
```

Comment:
On line 162 in this file, pipeReader is being closed in CopyObject handler - I think we need to do it deferred in line 150, so that in case of errors also the pipe is closed and the go routine returns. [I am not able to comment on that part of the code as there are no changes near there in this PR]

Reply:
> On line 162 in this file, pipeReader is being closed in CopyObject handler - I think we need to do it deferred in line 150, so that in case of errors also the pipe is closed and the go routine returns. [I am not able to comment on that part of the code as there are no changes near there in this PR]

I am removing the stale code which is doing a real object copy in xlObjects.CopyObject(). Because currently xl-sets is doing the actual copy and xlObjects.CopyObject() is only called to update metadata